Update `dynamic-range-limit` for latest spec changes

weinig opened the pull request:
#### e9250a6a45496869d52fadb9116333b3750a82a7
<pre>
Update `dynamic-range-limit` for latest spec changes
<a href="https://bugs.webkit.org/show_bug.cgi?id=287710">https://bugs.webkit.org/show_bug.cgi?id=287710</a>

Reviewed by Tim Nguyen, Darin Adler, and Gerald Squelart.

Two minor changes to the `dynamic-range-limit` spec:
    - `high` has been renamed `no-limit`.
    - `dynamic-range-limit-mix()` now has a lower bound minimum of two elements.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/inheritance.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/interpolation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/parsing.html:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/values/color/CSSDynamicRangeLimit.cpp:
* Source/WebCore/css/values/color/CSSDynamicRangeLimit.h:
* Source/WebCore/css/values/color/CSSDynamicRangeLimitMix.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.h:
* Source/WebCore/style/values/color/StyleDynamicRangeLimitMix.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimitMix.h:

Canonical link: <a href="https://commits.webkit.org/290515@main">https://commits.webkit.org/290515@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10ab23cdbf6f1527889bf14c2de250f9cabc76f8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90304 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9833 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45228 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95308 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/41080 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92356 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/10221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18150 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/69511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/41080 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93305 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/10221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/81917 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49872 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/10221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/36285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40211 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/10221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37352 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97132 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17492 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12861 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/78513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/17749 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77747 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77724 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19189 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20786 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/10751 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17502 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/22829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/17243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/20695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/19027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please make sure to export the WPT changes

squelart reviewed (approved):
Thank you!

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -66,7 +66,7 @@ template<typename> inline constexpr ASCIILiteral SerializationSeparator = ""_s;
         wrapped value;                                                        \
         template<typename... Args>                                            \
-        wrapper(Args&&... args) requires (requires { { wrapped(args...) }; }) \
+        wrapper(Args&&... args)                                               \
```
I don't understand this change nor does the change log explain

weinig replied:
Oh, I don't plan to land that. I was trying to work out how to fix the build issue some of the bots were seeing but I don't see locally (likely just an older clang).

The solution ended up being a slightly annoying extra type in this call:

```
            CSS::DynamicRangeLimitMixFunctionValue { CSS::DynamicRangeLimitMixParameters { WTFMove(resultBuilder) } }
```

weinig committed:
Update `dynamic-range-limit` for latest spec changes
https://bugs.webkit.org/show_bug.cgi?id=287710

Reviewed by Tim Nguyen, Darin Adler, and Gerald Squelart.

Two minor changes to the `dynamic-range-limit` spec:
    - `high` has been renamed `no-limit`.
    - `dynamic-range-limit-mix()` now has a lower bound minimum of two elements.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/inheritance.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/interpolation.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color-hdr/parsing.html:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/values/color/CSSDynamicRangeLimit.cpp:
* Source/WebCore/css/values/color/CSSDynamicRangeLimit.h:
* Source/WebCore/css/values/color/CSSDynamicRangeLimitMix.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.h:
* Source/WebCore/style/values/color/StyleDynamicRangeLimitMix.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimitMix.h:

Canonical link: https://commits.webkit.org/290515@main
